chainstate/launcher/src/lib.rs

```diff
@@ -19,25 +19,38 @@ mod config;
 
 use std::sync::Arc;
 
+use chainstate::InitializationError;
+use chainstate_storage::{BlockchainStorageRead, BlockchainStorageWrite};
+use storage_lmdb::resize_callback::MapResizeCallback;
+use utils::ensure;
+
 // Some useful reexports
 pub use chainstate::{
     chainstate_interface::ChainstateInterface, ChainstateConfig, ChainstateError as Error,
     DefaultTransactionVerificationStrategy,
 };
 pub use common::chain::ChainConfig;
 pub use config::{ChainstateLauncherConfig, StorageBackendConfig};
-use storage_lmdb::resize_callback::MapResizeCallback;
 
 /// Subdirectory under `datadir` where LMDB chainstate database is placed
 pub const SUBDIRECTORY_LMDB: &str = "chainstate-lmdb";
 
+const CHAINSTATE_STORAGE_VERSION_UNINITIALIZED: u32 = 0;
+const CHAINSTATE_STORAGE_VERSION_V1: u32 = 1;
+const CURRENT_CHAINSTATE_STORAGE_VERSION: u32 = CHAINSTATE_STORAGE_VERSION_V1;
+
 fn make_chainstate_and_storage_impl<B: 'static + storage::Backend>(
     storage_backend: B,
     chain_config: Arc<ChainConfig>,
     chainstate_config: ChainstateConfig,
 ) -> Result<Box<dyn ChainstateInterface>, Error> {
-    let storage = chainstate_storage::Store::new(storage_backend)
+    let mut storage = chainstate_storage::Store::new(storage_backend)
         .map_err(|e| Error::FailedToInitializeChainstate(e.into()))?;
+
+    check_storage_version(&mut storage)?;
+    check_magic_bytes(&mut storage, chain_config.as_ref())?;
+    check_chain_type(&mut storage, chain_config.as_ref())?;
```

Comment:
Perhaps move these into one call, `check_storage_compatibility()`, and move all these calls and impls into a submodules. Then we can test them in isolation.